### ISSUE TITLE
How to migrate data from one storage node to another

### ISSUE THREAD

**Issue by alexandnpu** [VictoriaMetrics/VictoriaMetrics#2290]
In my early VictoriaMetrics Cluster setup, some old linux boxes are used. These boxes' performance is not that good. So I decided to replace them with some new one.

But before I put the new one under production load, I want to migrate data from the old one first.
Is there some tool that I can use?

I read the doc about vmctl, but it seems that vmctl works with vmselect api. But how to do data migration with just one single storage node in the cluster mode?

Thanks

**Comment by hagen1778** [comment]
Hi @alexandnpu!
`vmctl` indeed does a `select=>insert` type of migration, including ability to filter selected data. It is useful for cases where you want to migrate only a fraction of data, or migrate data between cluster and single installations (since they're not replaceable), or re-shard data fron N shards to K shards.

In the cases where you need just to "move" data from single-to-single, or from vmstorage-to-vmstorage - consider just to copy data folder as is via any tool you like. For example, `rsync` is a good utility to do this.

**Comment by alexandnpu** [comment]
@hagen1778 , thanks for your reply. Folder copying is really a nice way for me!

**Comment by thislyc** [comment]
Hello, I have configured - replicationFactory=2. There are two vmstorage nodes in total, and one node needs to be migrated. I used rsync to migrate the files of a vmstorage node to a new node, but the data in the migration process was lost. Among the latest two nodes, one data is complete, and the other is missing half an hour of data. As expected, complete records should be returned, But the current situation is that the complete data is displayed at one time, and the half hour is missing from the data returned at the other time @hagen1778
